File: 283-MoveZeroes/283-MoveZeroes.cpp
// Last updated: 07/04/2026, 11:39:40
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int zeroes = 0 ;
        int n = nums.size();
        for ( int i = 0;i<n;i++){
            if ( nums[zeroes] == 0 && nums[i] !=0){
                swap(nums[i],nums[zeroes]);
            }
            if ( nums[zeroes] !=0) zeroes++;
        }
        
    }
};

File: 283-MoveZeroes/283-MoveZeroes.py
# Last updated: 07/04/2026, 11:39:44
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        
        
        
        """
        zero = 0 
        for i in range(len(nums)):
            
            if nums[zero] == 0 and nums[i] != 0 :
                nums[zero], nums[i] = nums[i] , nums[zero]
            if nums[zero] != 0 :
                zero +=1
                
        return nums

                
                
                
            
                
        